src/Symfony/Bridge/Doctrine/Tests/TestRepositoryFactory.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Tests;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\ObjectRepository;
+use Symfony\Bridge\Doctrine\Test\TestRepositoryFactory as TestTestRepositoryFactory;
+
+/**
+ * @author Andreas Braun <alcaeus@alcaeus.org>
+ */
+final class TestRepositoryFactory extends TestTestRepositoryFactory
+{
+    public function getRepository(EntityManagerInterface $entityManager, $entityName, bool $deprecate = false)
+    {
+        return parent::getRepository($entityManager, $entityName, $deprecate);
+    }
+
+    public function setRepository(EntityManagerInterface $entityManager, string $entityName, ObjectRepository $repository, bool $deprecate = false)
+    {
+        return parent::setRepository($entityManager, $entityName, $repository, $deprecate);
+    }
```

Comment:
```suggestion
    public function getRepository(EntityManagerInterface $entityManager, $entityName): ObjectRepository
    {
        return parent::getRepository($entityManager, $entityName, false);
    }

    public function setRepository(EntityManagerInterface $entityManager, string $entityName, ObjectRepository $repository): void
    {
        parent::setRepository($entityManager, $entityName, $repository, false);
    }
```

Reply:
Unfortunately, PHP throws an error when you don't add the `$deprecate` argument: https://3v4l.org/8JLgh (not sure why, to me the signature is still compatible when I omit an argument..)

Reply:
Oh right, because you have actually added the `$deprecate` argument upstream. I believe we don't want to have that argument in the final method signature, so can we comment the argument out like you did with`DoctrineTestHelper`?

As an alternative to the `$deprecate` argument, you could check if `static::class` points to the new class and only trigger the deprecation if this is not the case.

Reply:
> As an alternative to the $deprecate argument, you could check if static::class points to the new class and only trigger the deprecation if this is not the case.

Nice idea, I've applied this solution.